Dockerfile

```diff
@@ -20,75 +20,67 @@ ARG COMMIT_SHA
 COPY ui /app
-RUN npm run build -- --base=$UI_PUBLIC_URL
+WORKDIR /app
+RUN npm ci --omit=dev --ignore-scripts \
+  # Build frontend
+  && npm run build -- --base=$UI_PUBLIC_URL
 
 ### Build API ###
 FROM --platform=${BUILDPLATFORM:-linux/amd64} debian:bookworm AS api
 ARG TARGETPLATFORM
 
 COPY docker/install_build_dependencies.sh /tmp/
-RUN chmod +x /tmp/install_build_dependencies.sh && /tmp/install_build_dependencies.sh
-
 COPY docker/go_wrapper.sh /go/bin/go
-RUN chmod +x /go/bin/go
+COPY api /app
+WORKDIR /app
+
 ENV GOPATH="/go"
 ENV PATH="${GOPATH}/bin:${PATH}"
-
 ENV CGO_ENABLED 1
 
-RUN go env
-
-RUN mkdir -p /app
-WORKDIR /app
-
 # Download dependencies
-COPY api/go.mod api/go.sum /app/
-RUN go mod download
-
-# Patch go-face
-RUN sed -i 's/-march=native//g' ${GOPATH}/pkg/mod/github.com/!kagami/go-face*/face.go
-
-# Build dependencies that use CGO
-RUN go install \
-  github.com/mattn/go-sqlite3 \
-  github.com/Kagami/go-face
-
-# Copy and build api source
-COPY api /app
-RUN go build -v -o photoview .
+RUN chmod +x /tmp/install_build_dependencies.sh \
+  && chmod +x /go/bin/go \
+  && /tmp/install_build_dependencies.sh \
+  && go env \
+  && go mod download \
+  # Patch go-face
+  && sed -i 's/-march=native//g' ${GOPATH}/pkg/mod/github.com/!kagami/go-face*/face.go \
+  # Build dependencies that use CGO
+  && go install \
+    github.com/mattn/go-sqlite3 \
+    github.com/Kagami/go-face \
+  # Build api source
+  && go build -v -o photoview .
 
 ### Copy api and ui to production environment ###
-FROM debian:bookworm
+FROM debian:bookworm-slim
 ARG TARGETPLATFORM
 WORKDIR /app
 
-COPY api/data /app/data
-
-RUN apt update \
+# Create a user to run Photoview server
+RUN groupadd -r photoview \
+  && useradd -r -g photoview photoview \
+  && mkdir -p /home/photoview \
+  && chown -R photoview:photoview /app /home/photoview \
   # Required dependencies
-  && apt install -y curl gpg libdlib19.1 ffmpeg exiftool libheif1
-
-# Install Darktable if building for a supported architecture
-RUN if [ "${TARGETPLATFORM}" = "linux/amd64" ] || [ "${TARGETPLATFORM}" = "linux/arm64" ]; then \
-  apt install -y darktable; fi
-
-# Remove build dependencies and cleanup
-RUN apt purge -y gpg \
+  && apt update \
+  && apt install -y curl gpg libdlib19.1 ffmpeg exiftool libheif1 \
+  # Install Darktable if building for a supported architecture
+  && if [ "${TARGETPLATFORM}" = "linux/amd64" ] || [ "${TARGETPLATFORM}" = "linux/arm64" ]; then \
+    apt install -y darktable; \
+  fi \
+  # Remove build dependencies and cleanup
+  && apt purge -y gpg \
   && apt autoremove -y \
   && apt clean \
   && rm -rf /var/lib/apt/lists/*
 
-COPY --from=ui /app/dist /ui
-COPY --from=api /app/photoview /app/photoview
+COPY --chown=photoveiw:photoveiw api/data /app/data
+COPY --chown=photoveiw:photoveiw --from=ui /app/dist /ui
+COPY --chown=photoveiw:photoveiw --from=api /app/photoview /app/photoview
 
 ENV PHOTOVIEW_LISTEN_IP 127.0.0.1
 ENV PHOTOVIEW_LISTEN_PORT 80
```

Comment:
As a side-note, the default here should probably be changed to `8080` if we want it to work out-of-the-box without the need to grant any special capabilities (and don't forget about the `EXPOSE` below in that case).

Reply:
We provide a web-server app, so its default HTTP port should be 80. Then a user can safely map it to whatever free port, he\she wants.
We provide the `docker-compose.yml` with the port mapping and document the procedure of the initial setup. Also, the port 8080 might be used as well, so I don't see a reason for this change.

Reply:
Port 80 is a privileged port. It either requires root access or assignment of CAP_NET_BIND_SERVICE. You can not use this port otherwise. What port you use outside of the container doesn't matter, that can be w/e you want and does not need to be changed.

Reply:
To clarify, there's 2 layers of ports here. The compose file defines a port mapping for Docker; you're asking your container runtime to open port X and forward it to port Y inside the container. What port X is doesn't matter and is up to the user as long as their container runtime has the permissions to bind the chosen port. This is about port Y, the one inside the container. Since the container is not run with root or assigned special capabilities, it can not bind port 80. I.o.w., without either as a user deciding to run as root after all, assigning CAP_NET_BIND_SERVICE to the container or setting PHOTOVIEW_LISTEN_PORT manually to something above 1024, this will not run out-of-the-box if it's not changed.

I suggested port 8080 for this value as that is generally the most commonly used port for non-privileged HTTP but anything above 1024 is technically fine.

Reply:
> Port 80 is a privileged port. It either requires root access or assignment of the CAP_NET_BIND_SERVICE capability. You can not use this port otherwise. What port you use outside of the container doesn't matter, that can be w/e you want and does not need to be changed.

Ok, so what is the problem? Sorry, I still didn't get the point)
Now the app is successfully run under the `photoview` user with 80 port internally. Why should we change it?

Reply:
https://github.com/photoview/photoview/pull/863#discussion_r1576323967  

It might be that the container runtime and/or the version of a given runtime has some impact on this as well. I believe Docker at some point changed its default configuration to allow the use of privileged ports by containers but I don't know in which version, nor of any other container runtime that allows this, hence the above note.